### PR TITLE
Add timeout for InstallPlan creation in install_operator

### PR DESCRIPTION
**Pull request by adolfo-ab**
##### Short description:
Add specific timeout for InstallPlan creation in install_operator function

##### More details:
Right now, we only have a timeout to wait for InstallPlan status to be complete. In this PR, a new timeout is added to wait for InstallPlan to be created, without altering existing functionality.

## Summary by CodeRabbit

- **New Features**
	- Enhanced timeout management for operator installations with separate timeouts for creation and status completion.
	- Introduced warnings for deprecated timeout usage to guide users towards new parameters.

- **Documentation**
	- Updated function documentation to clarify new parameters and their purposes.

### CODE REVIEW

File: ocp_utilities/operators.py [hunk 1]
```diff
@@ -153,6 +158,7 @@ def install_operator(
     source: str = "",
     install_plan_approval: str = "Automatic",
     starting_csv: str = "",
+    ip_creation_timeout: int = TIMEOUT_10MIN,
```
[the remark this answers is not in the document]
Comment (adolfo-ab) [reply ocp_utilities/operators.py:161]: Cannot rename `timeout` argument without altering current behavior and affecting other projects

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_utilities/operators.py [hunk 2]
```diff
@@ -67,23 +67,28 @@ def wait_for_install_plan_from_subscription(
 
 
 def wait_for_operator_install(
-    admin_client: DynamicClient, subscription: Subscription, timeout: int = TIMEOUT_5MIN
+    admin_client: DynamicClient,
+    subscription: Subscription,
+    creation_timeout: int = TIMEOUT_5MIN,
+    status_complete_timeout: int = TIMEOUT_5MIN,
```
[the remark this answers is not in the document]
Comment (myakove) [reply ocp_utilities/operators.py:73]: We also need to deprecate `timeout` in the next major release in favor of `creation_timeout` and `status_complete_timeout`

[the remark this answers is not in the document]
Comment (myakove) [reply ocp_utilities/operators.py:73]: `if timeout != TIMEOUT_5MIN:` is not the best way to deprecate this.
Users can still pass `timeout` with the default timeout value.

In this case, the deprecation warning will not show

[the remark this answers is not in the document]
Comment (adolfo-ab) [reply ocp_utilities/operators.py]: Done

File: ocp_utilities/operators.py [hunk 3]
```diff
@@ -67,23 +68,43 @@ def wait_for_install_plan_from_subscription(
 
 
 def wait_for_operator_install(
-    admin_client: DynamicClient, subscription: Subscription, timeout: int = TIMEOUT_5MIN
-) -> None:
+    admin_client: DynamicClient,
+    subscription: Subscription,
+    timeout: Optional[int] = None,
+    creation_timeout: Optional[int] = None,
```
Comment (rnetser) [inline ocp_utilities/operators.py:74]:
```suggestion
    insall_plan_creation_timeout: Optional[int] = None,
```

File: ocp_utilities/operators.py [hunk 4]
```diff
@@ -67,23 +68,43 @@ def wait_for_install_plan_from_subscription(
...
+    status_complete_timeout: Optional[int] = None,
```
Comment (rnetser) [inline ocp_utilities/operators.py:75]:
```suggestion
    install_plan_status_complete_timeout: Optional[int] = None,
```

File: ocp_utilities/operators.py [hunk 5]
```diff
@@ -153,6 +174,7 @@ def install_operator(
     source: str = "",
     install_plan_approval: str = "Automatic",
     starting_csv: str = "",
+    ip_creation_timeout: int = TIMEOUT_10MIN,
```
Comment (rnetser) [inline ocp_utilities/operators.py:177]:
```suggestion
    install_plan_creation_timeout: int = TIMEOUT_10MIN,
```

File: ocp_utilities/operators.py [hunk 6]
```diff
@@ -67,23 +68,43 @@ def wait_for_install_plan_from_subscription(
...
     """
     Wait for the operator to be installed, including InstallPlan and CSV ready.
 
     Args:
         admin_client (DynamicClient): Cluster client.
         subscription (Subscription): Subscription instance.
-        timeout (int): Timeout in seconds to wait for operator to be installed.
+        timeout (int, optional): [Deprecated] Timeout in seconds used if specific timeouts are not provided.
+        creation_timeout (int, optional): Timeout in seconds to wait for InstallPlan to be created.
+        status_complete_timeout (int, optional): Timeout in seconds to wait for operator to be installed
```
Comment (rnetser) [inline ocp_utilities/operators.py:85]: this timeout for ip status; there's a wait for csv after

File: ocp_utilities/operators.py [hunk 7]
```diff
@@ -174,7 +196,8 @@ def install_operator(
         starting_csv (str, optional): The specific CSV to start from.
         target_namespaces (list, optional): Target namespaces for the operator install process.
             If not provided, a namespace with te operator name will be created and used.
-        timeout (int): Timeout in seconds to wait for operator to be ready.
+        ip_creation_timeout (int): Timeout in seconds to wait for InstallPlan to be created.
+        timeout (int): Timeout in seconds to wait for InstallPlan status to be COMPLETE.
```
Comment (rnetser) [inline ocp_utilities/operators.py:200]: we should have the same logic as `wait_for_operator_install` here.

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_utilities/operators.py [hunk 8]
```diff
@@ -71,7 +71,7 @@ def wait_for_operator_install(
     admin_client: DynamicClient,
     subscription: Subscription,
     timeout: Optional[int] = None,
-    insall_plan_creation_timeout: Optional[int] = None,
+    install_plan_creation_timeout: Optional[int] = None,
```
Comment (dbasunag) [inline ocp_utilities/operators.py:74]: Doc header says `creation_timeout`, is this a typo?

**Pull request closed without merging**